cypress/integration/consents.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable cypress/no-unnecessary-waiting */
+/* eslint-disable require-jsdoc */
+var Countly = require("../../lib/countly");
+var hp = require("../support/helper");
+
+function initMain() {
+    Countly.init({
+        app_key: "YOUR_APP_KEY",
+        url: "https://try.count.ly",
+        require_consent: true,
+        device_id: "György Ligeti",
+        tests: true,
+        max_events: -1,
+        debug: true
+    });
+}
+
+// gathered events to add
+function events() {
+    Countly.add_event({
+        key: "a",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+    Countly.add_event({
+        key: "[CLY]_view",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+    Countly.add_event({
+        key: "[CLY]_nps",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+    Countly.add_event({
+        key: "[CLY]_survey",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+    Countly.add_event({
+        key: "[CLY]_star_rating",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+    Countly.add_event({
+        key: "[CLY]_orientation",
+        count: 1,
+        segmentation: {
+            "custom key": "custom value",
+        },
+    });
+}
+// tests
+describe("Consent tests", () => {
```

Comment:
Missing a test with all consents and then recording all events.
And no consent required and again recording all